autometa/config/databases.py

```diff
@@ -367,26 +390,26 @@ def press_hmms(self):
         NoneType
 
         """
-        hmm_search_str = os.path.join(self.markers_dir, '*.h3?')
+        hmm_search_str = os.path.join(self.markers_dir, "*.h3?")
         # First search for pressed hmms to remove from list to hmmpress
-        pressed_hmms = set(
+        pressed_hmms = {
             os.path.realpath(os.path.splitext(fp)[0])
             for fp in glob(hmm_search_str)
-            if not fp.endswith('.md5')
-        )
+            if not fp.endswith(".md5")
+        }
```

Comment:
I'm not sure I understand what is going on here. Instead of being cast as a `set`, the filepaths not ending in `'.md5'` are cast as a `dict`? Why is this change needed, and does it just become a dictionary with keys and no values?

Reply:
`pressed_hmms` is a set. This is an example of a set comprehension. Instead of constructing a list then casting the list as a set, a set is simply constructed.

The syntax for dictionary or set comprehensions are similar to list comprehensions.

```python
list_comprehension = [elem for elem in elems]
# Notice curly braces
set_comprehension = {elem for elem in elems}
# Notice curly braces _and_ the colon to separe the key:value pairs
dict_comprehension = {i: elem for i, elem in enumerate(elems)}

```

Reply:
Yeah, sorry. For whatever reason I have not seen that before.